persistence/src/main/java/com/lightbend/lagom/javadsl/persistence/cassandra/CassandraContactPoint.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package com.lightbend.lagom.javadsl.persistence.cassandra;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Preconditions;
+import java.net.URI;
+import javax.annotation.Nullable;
+import javax.annotation.ParametersAreNonnullByDefault;
+import javax.annotation.concurrent.Immutable;
+ 
+@ParametersAreNonnullByDefault
+@Immutable
+public final class CassandraContactPoint {
```

Comment:
Why does this have `@Immutable` annotation but also has implementation? Can we use immutables to generate the implementation?